nexus/types/src/quiesce.rs

```diff
@@ -151,26 +157,77 @@ impl SagaQuiesceHandle {
         SagaQuiesceHandle { log, inner }
     }
 
-    /// Disallow new sagas from being started or re-assigned to this Nexus
+    /// Set the intended quiescing state
+    ///
+    /// Quiescing is currently a one-way trip.  Once we start quiescing, we
+    /// cannot then re-enable sagas.
+    pub fn set_quiescing(&self, quiescing: bool) {
+        self.inner.send_if_modified(|q| {
+            let new_state = if quiescing {
+                SagasAllowed::DisallowedQuiesce
+            } else {
+                SagasAllowed::Allowed
+            };
+
+            match q.new_sagas_allowed {
+                SagasAllowed::DisallowedUnknown => {
+                    info!(
+                        &self.log,
+                        "initial quiesce state";
+                        "initial_state" => ?new_state
+                    );
+                    q.new_sagas_allowed = new_state;
+                    true
+                }
+                SagasAllowed::Allowed if quiescing => {
+                    info!(&self.log, "saga quiesce starting");
+                    q.new_sagas_allowed = SagasAllowed::DisallowedQuiesce;
+                    true
+                }
+                SagasAllowed::DisallowedQuiesce if !quiescing => {
+                    // This should be impossible.  Report a problem.
+                    error!(
+                        &self.log,
+                        "asked to stop quiescing after previously quiescing"
+                    );
+                    false
+                }
+                _ => {
```

Comment:
I wonder if this match would be clearer if we nested the `if`s in the two branches above, which (I think?) would let us drop this `_` entirely; e.g.,

```rust
SagasAllowed::Allowed => { 
    // If sagas are currently allowed but we need to quiesce, switch to disallowing them.
    if quiescing {
        // ... all the stuff from above ...
    }
}

Reply:
I think I just really dislike `_` and will take nearly any excuse to drop it. :joy: